container/docker/factory.go

```diff
@@ -62,6 +63,10 @@ func UseSystemd() bool {
 	return useSystemd
 }
 
+func RootDir() string {
+	return *dockerRootDir
```

Comment:
I did some refactoring to get it from docker factory. The docker base dir is chosen by -g flag, but none of the docker APIs return the path they are using. We can either parse docker command line, or infer it from docker client by parsing graph driver info. Easiest way would be to enhance docker to put the base dir somewhere.

I tried doing it in docker factory, but it was too ugly. Will revisit soon.